src/chat/__tests__/chatReducers-test.js

```diff
@@ -656,5 +656,25 @@ describe('chatReducers', () => {
 
       expect(newState).toEqual(expectedState);
     });
+
+    test('when replaceExisting is true, common messages are not replaced', () => {
+      const commonMessages = [{ id: 2, timestamp: 4 }, { id: 3, timestamp: 5 }];
+      const initialState = deepFreeze({
+        messages: {
+          [homeNarrowStr]: [{ id: 1, timestamp: 3 }, ...commonMessages],
+        },
+      });
+
+      const action = deepFreeze({
+        type: MESSAGE_FETCH_SUCCESS,
+        narrow: [],
+        messages: [{ id: 2, timestamp: 4 }, { id: 3, timestamp: 5 }],
+        replaceExisting: true,
+      });
+
+      const newState = chatReducers(initialState, action);
+
+      expect(...newState.messages[homeNarrowStr]).toBe(...commonMessages);
```

Comment:
Sure, before making this change just wanna add this
Here, I intended to use a referential comparison over messages
if we use `toEqual`, even the following will pass the test case
```javascript
      const differentArr = [{ id: 2, timestamp: 4 }, { id: 3, timestamp: 5 }];

      expect(...newState.messages[homeNarrowStr]).toEqual(...differentArr);
```

and if we use `toBe`, the above test case would fail making sure that we maintain referential equality over common messages (and thus prevent needless re-renders)